cmd/edgectl/aes_install.go

```diff
@@ -200,13 +200,24 @@ func (i *Installer) loopUntil(what string, how func() error, lc *loopConfig) err
 // the Pod is Running (though not necessarily Ready). This should be good enough
 // to report the "deploy" status to metrics.
 func (i *Installer) GrabAESInstallID() error {
-	// Note: This is brittle. If there is more than one pod, this returns all
-	// the pod names squashed together. This should not occur in normal usage,
-	// i.e. when installing a single version of AES, maybe repeatedly.
-	pod, err := i.CaptureKubectl("get AES pod", "", "-n", "ambassador", "get", "pods", "-l", "service=ambassador", "-o", "go-template={{range .items}}{{.metadata.name}}{{end}}")
+	// Find an AES Pod
+	aesImage := "quay.io/datawire/aes:" + i.version
+	pod := ""
+	podInterface := i.coreClient.Pods("ambassador") // namespace
+	pods, err := podInterface.List(k8sTypesMetaV1.ListOptions{})
 	if err != nil {
 		return err
 	}
+	for _, k8sPod := range pods.Items {
+		containers := k8sPod.Spec.Containers
+		if len(containers) != 1 || containers[0].Image != aesImage || containers[0].Command != nil {
```

Comment:
I wonder if we could have a looser conditional here? I'd go with a simple `if containers[0].Image != aesImage` {}

In case we eventually package AES with a sidecar container or change the docker entrypoint to remove the command, it would introduce a breaking change in the installer.

Reply:
Here are the missing comments. :slightly_frowning_face: 
- `len(containers) != 1` is to skip the Traffic Agent sidecar
- `containers[0].Command != nil` is to skip the Traffic Manager

Reply:
Added.

Reply:
Maybe iterating over the containers would be a better option then. Otherwise, if we run a single pod with the traffic agent, we'll be skipping over it and never will know the install_id!

Reply:
Hmm? I'm confused. Can you explain what case we're messing up?

Reply:
We synced elsewhere. I will update this PR to check for the Traffic Agent in a different way, rather than relying on `len(containers) == 1`, as that will break as soon as new Istio stuff lands, or if anything else injects a container alongside Ambassador.